### PR TITLE
SyncUser.all() returns Map

### PR DESCRIPTION
**Pull request by dalinaum**
For #3697

### CODE REVIEW

File: realm/realm-library/src/objectServer/java/io/realm/SyncUser.java [hunk 1]
```diff
@@ -106,18 +107,18 @@ public static SyncUser currentUser() {
...
-    public static Collection<SyncUser> all() {
+    public static Map<String, SyncUser> all() {
         UserStore userStore = SyncManager.getUserStore();
         Collection<SyncUser> storedUsers = userStore.allUsers();
-        List<SyncUser> result = new ArrayList<SyncUser>(storedUsers.size());
+        Map<String, SyncUser> map = new HashMap<String, SyncUser>();
         for (SyncUser user : storedUsers) {
             if (user.isValid()) {
-                result.add(user);
+                map.put(user.getIdentity(), user);
```
Comment (kneth) [inline realm/realm-library/src/objectServer/java/io/realm/SyncUser.java]: https://github.com/realm/realm-js/blob/master/src/js_sync.hpp#L130 is skipping admin users.

**Comment by dalinaum** [comment]
Please review this @realm/java

File: realm/realm-library/src/objectServer/java/io/realm/SyncUser.java (hunk 1 above)
Comment (beeender) [inline realm/realm-library/src/objectServer/java/io/realm/SyncUser.java]: No clue why does js filters out the admin user....

Comment (dalinaum) [inline realm/realm-library/src/objectServer/java/io/realm/SyncUser.java]: @beeender Although I don't know why js filters out admin users, it seems that Object store already excludes them.

File: realm/realm-library/src/objectServer/java/io/realm/SyncUser.java [hunk 2]
```diff
@@ -106,18 +107,18 @@ public static SyncUser currentUser() {
      * Returns all valid users known by this device.
      * A user is invalidated when he/she logs out or the user's access token expires.
      *
-     * @return a list of all known valid users.
+     * @return a map of all known valid users.
```
Comment (cmelchior) [inline realm/realm-library/src/objectServer/java/io/realm/SyncUser.java]: Also document that the key is the users identifier.

**Pull request merged**